src/projectscene/view/clipsview/clipslistmodel.cpp

```diff
@@ -440,6 +440,10 @@ bool ClipsListModel::trimLeftClip(const ClipKey& key, bool completed)
         return false;
     }
 
+    if (muse::RealIsEqualOrLess(newStartTime, 0.0)) {
+        newStartTime = 0.0;
```

Comment:
I think it would be useful to move this adjustment above the check with `item->clip().startTime`, and I also recommend using std::max